usecases/scenario_publications_usecase.go

```diff
@@ -114,62 +114,118 @@ func (usecase *ScenarioPublicationUsecase) ExecuteScenarioPublicationAction(
+		status.PreparationStatus = models.PreparationStatusRequired
+	}
+
+	if numPending == 0 {
+		status.PreparationServiceStatus = models.PreparationServiceStatusAvailable
+	} else {
+		status.PreparationServiceStatus = models.PreparationServiceStatusOccupied
+	}
+
+	return
+}
+
+func (usecase *ScenarioPublicationUsecase) StartPublicationPreparation(
+	ctx context.Context,
+	scenarioIterationId string,
+) error {
+	logger := utils.LoggerFromContext(ctx)
+
+	indexesToCreate, numPending, err := usecase.getIndexesToCreate(ctx, scenarioIterationId)
 	if err != nil {
-		return false, err
+		return errors.Wrap(err, "Error while fetching indexes to create in StartPublicationPreparation")
+	}
+
+	if len(indexesToCreate) == 0 {
+		return nil
+	}
+
+	if numPending > 0 {
+		return errors.Wrap(
+			models.ConflictError, // return 409 if the db is busy creating indexes in this schema
+			"There are still pending indexes in the schema")
 	}
 
 	organizationId, err := usecase.OrganizationIdOfContext()
 	if err != nil {
-		return false, err
+		return err
 	}
-	scenarios, err := usecase.scenarioListRepository.ListScenariosOfOrganization(ctx, exec, organizationId)
+	db, err := usecase.executorFactory.NewClientDbExecutor(ctx, organizationId)
 	if err != nil {
-		return false, err
-	}
-	liveScenarios := utils.Filter(scenarios, func(scenario models.Scenario) bool {
-		return scenario.LiveVersionID != nil
-	})
-	activeScenarioIterations, err := pure_utils.MapErr(liveScenarios, func(
-		scenario models.Scenario,
-	) (models.ScenarioIteration, error) {
-		it, err := usecase.scenarioFetcher.FetchScenarioAndIteration(ctx, exec, *scenario.LiveVersionID)
-		if err != nil {
-			return models.ScenarioIteration{}, err
-		}
-		return it.Iteration, nil
-	})
+		return errors.Wrap(
+			err,
+			"Error while creating client schema executor in StartPublicationPreparation")
+	}
+	num, err := usecase.ingestedDataIndexesRepository.CreateIndexesAsync(ctx, db, indexesToCreate)
+	if err != nil {
+		return errors.Wrap(err, "Error while creating indexes in StartPublicationPreparation")
+	}
+	logger.InfoContext(
+		ctx,
+		fmt.Sprintf("%d indexes pending creation in: %+v\n", num, indexesToCreate), "org_id", organizationId,
+	)
+	return nil
+}
+
+func (usecase *ScenarioPublicationUsecase) getIndexesToCreate(
+	ctx context.Context,
+	scenarioIterationId string,
+) (toCreate []models.ConcreteIndex, numPending int, err error) {
+	exec := usecase.executorFactory.NewExecutor()
+	iterationToActivate, err := usecase.scenarioFetcher.FetchScenarioAndIteration(ctx, exec, scenarioIterationId)
 	if err != nil {
-		return false, errors.Wrap(err, "Error while fetching active scenario iterations in CreateDatamodelIndexesForScenarioPublication")
+		return toCreate, numPending, err
+	}
+	if err := usecase.enforceSecurity.PublishScenario(iterationToActivate.Scenario); err != nil {
```

Comment:
Open question: is this the right permission also for getting the status ? 
At least, we'll have to make sure we don't call it with non publisher users if so